src/map/script.c

```diff
@@ -8623,6 +8622,34 @@ BUILDIN(readparam) {
 	return true;
 }
 
+/*==============================================*
+ * setparam(<parameter number>, <value>{, "<Player Name>" | <account id> });
+ * Uses pc->setparam() on a specified Account ID. [Wolfie]
+ *==============================================*/
+BUILDIN(setparam)
+{
+	struct script_data *data = script_getdata(st, 2);
+	int type, val = script_getnum(st, 3);
+	struct map_session_data *sd;
+
+	if (reference_toparam(data))
+		type = reference_getparamtype(data);
+	else
+		type = script->conv_num(st, data);
```

Comment:
hmm ... well since you are responsible for this PR,
might as well just make the change here without open another Issue

this kind of check miss out `reference_tovariable` check,
that's why the `asdf` can get through into the pc_setparam function
also add `reference_toconstant` check while at it

Reply:
`reference_tovariable` is just a macro for `script->str_data[reference_getid(data)].type == C_NAME` and for `reference_toconstant` it is `script->str_data[reference_getid(data)].type == C_INT`, while for `reference_toparam` it is `script->str_data[reference_getid(data)].type == C_PARAM`.

Because `type` cannot `== C_NAME == C_INT == C_PARAM` it makes this check redundant, since `C_PARAM` ensures it can't be anything else.

The correct check would be `data_isreference(data) && reference_toparam(data)`